File: Python3/0448_find_all_numbers_disappeared_in_an_array.py
# https://leetcode.cn/problems/find-all-numbers-disappeared-in-an-array/

class Solution:
    def findDisappearedNumbers(self, nums):
        """
        Inputs:
            nums: list[int]
        Outputs:
            res: list[int]
        """
        n = len(nums)
        cnt = [0 for _ in range(n)]
        for num in nums:
            cnt[num-1] = 1
        
        res = []
        for i in range(n):
            if cnt[i] == 0:
                res.append(i+1)
        return res

File: Cpp/0448_find_all_numbers_disappeared_in_an_array.cpp
// https://leetcode.cn/problems/find-all-numbers-disappeared-in-an-array/

class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int n = nums.size();
        vector<int> cnt(n, 0);
        for (int i: nums) {
            cnt[i-1] = 1;
        }
        vector<int> res;
        for (int i = 0; i < n; ++i) {
            if (cnt[i] == 0) {
                res.push_back(i+1);
            }
        }
        return res;
    }
};
